### PR TITLE
[PDI-17667] Use of vulnerable component jetty 8.1.15, CVE-2017-9735, …

### PR DESCRIPTION
**Pull request by smmribeiro**
…CVE-2017-7657 CVE-2015-2080

[PPP-4183] Use of Vulnerable Component: Multiple Jetty Components [Listed in Description] (CVE-2017-7656 | CVE-2017-7657 | CVE-2017-7658 | CVE-2015-2080 | CVE-2017-9735 )

Centralized Jetty components: use dependency management.
Cleared some unnecessary exclusions.
Some classes had to be updated to work the way Jetty now works.

Depends on pentaho/maven-parent-poms#121

@pentaho-lmartins @pentaho/tatooine
@graimundo @nantunes

### CODE REVIEW

File: engine/pom.xml [hunk 1]
```diff
@@ -760,19 +754,11 @@
...
-        </exclusion>
-      </exclusions>
+      <artifactId>jetty-jaas</artifactId>
     </dependency>
     <dependency>
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-server</artifactId>
-      <version>${jetty-server.version}</version>
       <exclusions>
         <exclusion>
```
Comment (graimundo) [inline engine/pom.xml]: @smmribeiro Does the new version still bring in these transitives? Meaning, do we still need this exclusions?

File: engine/pom.xml [hunk 2]
```diff
@@ -930,6 +916,18 @@
       <version>${xmlunit.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
+      <version>3.1.0</version>
+      <exclusions>
+        <exclusion>
+          <groupId>*</groupId>
```
Comment (graimundo) [inline engine/pom.xml]: @smmribeiro Why the exclusions?

Comment (smmribeiro) [reply engine/pom.xml]: @graimundo, to be honest I don't know/remember why I added those exclusions... even more being a test scoped dependency.
I'll verify it.

File: engine/pom.xml (hunk 1 above)
Comment (smmribeiro) [reply engine/pom.xml]: Nope, seems not!
I'll remove it.

File: engine/pom.xml (hunk 2 above)
Comment (smmribeiro) [reply engine/pom.xml]: Resolving as I've updated the code.

**Comment by smmribeiro** [comment]
Closing PR: to simplify Karaf's upgrade a different approach was defined!

**Pull request closed without merging**